### DISCUSSION TITLE
Need help with simple face detection and memory management

### DISCUSSION THREAD

**Discussion by NateFaulkenberry** [vladmandic/face-api#106] [Q&A] [answered]
Hi, I'm trying face-api for the first time, and while it detects faces successfully, I am running into a wall with memory management. I've looked through your examples and tried a bunch of different solutions, but for some reason my tensors don't seem to be disposed, and memory on my server gets eaten up quickly. Just by looking at my code could you point out where I might be going wrong? Also is this library meant to work with Typescript? I appreciate any help. Thank you!

```
import path from 'path'
import '@tensorflow/tfjs-node'
import * as faceapi from '@vladmandic/face-api'

(async () => {
    const models = path.resolve(process.cwd(), 'models')
    await faceapi.nets.tinyFaceDetector.loadFromDisk(models)
})()

// https://github.com/vladmandic/face-api
const detect = async (buffer) => {
    // @ts-ignore
    const tensor = faceapi.tf.node.decodeImage(buffer)

    const options = new faceapi.TinyFaceDetectorOptions()

    // @ts-ignore
    const faces = await faceapi.detectAllFaces(tensor, options)
    
    // @ts-ignore
    faceapi.tf.dispose(tensor)
    
    // @ts-ignore
    console.log(faceapi.tf.memory())

    return faces.map(face => {
        return {
            x: face.box.x,
            y: face.box.y,
            width: face.box.width,
            height: face.box.height,
            weight: 1.0
        }
    })
}

export { detect }
```

**Comment by vladmandic** [comment, marked as answer]
i just run your code in 1k loop, i don't see a tensor leak - can you show what your `console.log` output is?

but you do have one memory issue - you create new `options` object on each detect call - why? create `options` object just once at the time of model load.
